apps/api_web/lib/api_web/event_stream.ex

```diff
@@ -37,50 +38,52 @@ defmodule ApiWeb.EventStream do
     ensure_timer(%__MODULE__{conn: conn, pid: pid, timeout: timeout})
   end
 
-  @spec hibernate_loop(state) :: no_return | {:error, term}
+  @spec hibernate_loop(state) :: Plug.Conn.t()
   def hibernate_loop(state) do
     case receive_result(state) do
-      {:ok, state} ->
+      {:continue, state} ->
         :proc_lib.hibernate(__MODULE__, :hibernate_loop, [state])
 
-      error ->
+      {:close, conn} ->
         Supervisor.server_unsubscribe(state.pid)
-        error
+        conn
     end
   end
 
-  @spec receive_result(state) :: {:ok, state} | {:error, term}
-  def receive_result(state) do
+  @spec receive_result(state) :: {:continue, state} | {:close, Plug.Conn.t()} | {:error, term}
+  def receive_result(%{pid: pid} = state) do
     receive do
       {:events, events} ->
-        chunks =
-          for {type, item} <- events do
-            ["event: ", type, "\ndata: ", item, "\n\n"]
-          end
+        chunks = for {type, item} <- events, do: ["event: ", type, "\ndata: ", item, "\n\n"]
+        continue(state, chunks)
 
-        update_state(state, chunk(state.conn, chunks))
+      :timeout ->
+        continue(state, ": keep-alive\n")
 
       {:error, rendered} when is_list(rendered) ->
-        {:error, chunk(state.conn, ["event: error\ndata: ", rendered, "\n\n"])}
+        close(state, ["event: error\ndata: ", rendered, "\n\n"])
 
-      :timeout ->
-        update_state(state, chunk(state.conn, ": keep-alive\n"))
+      {:DOWN, _ref, :process, ^pid, _reason} ->
+        close(state)
```

Comment:
This should probably check the `reason` and close normally if it's a regular shutdown reason and send an error first otherwise.

Reply:
What should the `data` of the event be in this case? I don't think there's much we could meaningfully tell a consumer just from knowing the DiffServer crashed, but I think there has to be some `data`.

Reply:
Ah, never mind, I think we can easily render a JSON-API 500 error in a similar way as the DiffServer does errors.